packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/internal/ApiKeyAuthImpl.kt

```diff
@@ -0,0 +1,162 @@
+package io.realm.kotlin.mongodb.internal
+
+import io.realm.kotlin.internal.ObjectIdImpl
+import io.realm.kotlin.internal.interop.ObjectIdWrapper
+import io.realm.kotlin.internal.interop.RealmInterop
+import io.realm.kotlin.internal.interop.sync.ApiKeyWrapper
+import io.realm.kotlin.internal.platform.freeze
+import io.realm.kotlin.internal.util.use
+import io.realm.kotlin.mongodb.auth.ApiKey
+import io.realm.kotlin.mongodb.auth.ApiKeyAuth
+import io.realm.kotlin.mongodb.exceptions.AppException
+import io.realm.kotlin.mongodb.exceptions.ServiceException
+import io.realm.kotlin.types.ObjectId
+import kotlinx.coroutines.channels.Channel
+
+internal class ApiKeyAuthImpl(override val app: AppImpl, override val user: UserImpl) : ApiKeyAuth {
+
+    private fun unwrap(apiKeyData: ApiKeyWrapper): ApiKey {
+        return ApiKey(
+            ObjectIdImpl(apiKeyData.id),
+            apiKeyData.value,
+            apiKeyData.name,
+            !apiKeyData.disabled
+        )
+    }
+
+    override suspend fun create(name: String): ApiKey {
+        try {
+            Channel<Result<ApiKey>>(1).use { channel ->
+                RealmInterop.realm_app_user_apikey_provider_client_create_apikey(
```

Comment:
Similarly to the comment in `RealmInterop`, it looks like you could unify some of these parts under shared logic.